pkg/controller/operators/catalog/operator.go

```diff
@@ -1654,11 +1659,54 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				plan.Status.Plan[i].Status = status
 
 			default:
-				return v1alpha1.ErrInvalidInstallPlan
-			}
+				resource := olmdynamic.IsSupported(step.Resource.Kind)
+				if resource == nil {
+					// Not a supported resource
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusUnsupportedResource
+					return v1alpha1.ErrInvalidInstallPlan
+				}
+
+				// Entering unstructured territory... To infinity and beyond!
+
+				// Marshal the manifest into an unstructured object
+				dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(step.Resource.Manifest), 10)
+				unstructuredObject := &unstructured.Unstructured{}
+				if err := dec.Decode(unstructuredObject); err != nil {
+					return errorwrap.Wrapf(err, "error decoding %s object to an unstructured object: %s", step.Resource.Name)
+				}
 
+				var resourceInterface dynamic.ResourceInterface
+				if resource.IsNamespaced() == true {
+					ownerutil.AddOwner(unstructuredObject, plan, false, false)
+					unstructuredObject.SetNamespace(namespace)
+					resourceInterface, err = olmdynamic.NamespacedResourceInterface(o.dynamicClient, namespace, unstructuredObject)
+					if err != nil {
+						return err
+					}
+				} else {
+					resourceInterface, err = olmdynamic.ClusterResourceInterface(o.dynamicClient, unstructuredObject)
+					if err != nil {
+						return err
+					}
+				}
+
+				// Update UIDs on all CSV OwnerReferences
+				updated, err := o.getUpdatedOwnerReferences(unstructuredObject.GetOwnerReferences(), plan.Namespace)
+				if err != nil {
+					return errorwrap.Wrapf(err, "error generating ownerrefs for unstructured object: %s", unstructuredObject.GetName())
+				}
+				unstructuredObject.SetOwnerReferences(updated)
+
+				// Ensure Object
+				status, err := ensurer.EnsureUnstructuredObject(resourceInterface, unstructuredObject)
+				if err != nil {
+					return err
+				}
+
+				logrus.Info("Updating plan status")
+				plan.Status.Plan[i].Status = status
+			}
 		default:
```

Comment:
is this an extra default?

Reply:
There are two nested switch cases, this default is for the initial switch.